### DISCUSSION TITLE
Best approach for using mapped field?

### DISCUSSION THREAD

**Discussion by DLesage25** [prisma/prisma#10835] [Q&A]
I have a schema similar to the following:
```
model User {
  id                          Int                @id @default(autoincrement())
  firstName            String
  lastName             String
  preferredName   String?
}
```

My database connects to a series of products that need to display a User's name. The problem is, user's have `firstName` and `preferredName`, and the latter should take precedence always. Meaning, only if `preferredName` is not set, the user's firstName should be provided. Usually, if only one endpoint was reading from my DB I would build that logic into the endpoint, but in this case, there are multiple apps and services reading from my DB. I want to find a way to build this logic into another field called `displayName`.

Essentially, `displayName` would provide a user's preferredName, but default to the `firstName` if no preferred name is set. Is there a way to do this in Prisma without having to build a ton of custom logic into my endpoint/service?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
